app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb

```diff
@@ -27,9 +28,79 @@ def ems_inv_to_hashes
       @data
     end
 
+    def self.miq_template_type
+      "ManageIQ::Providers::Lenovo::PhysicalInfraManager::Template"
+    end
+
     private
 
     def get_physical_servers
+      nodes = all_server_resources
+
+      nodes = nodes.map do |node|
+        XClarityClient::Node.new node
+      end
+      process_collection(nodes, :physical_servers) { |node| parse_physical_server(node) }
+    end
+
+    def get_hardwares(node)
+      {:firmwares => get_firmwares(node.firmware)}
+    end
+
+    def get_firmwares(node)
+      firmwares = node.map do |firmware|
+        parse_firmware(firmware)
+      end
+      firmwares
+    end
+
+    def parse_firmware(firmware)
+      {
+        :name         => "#{firmware["role"]} #{firmware["name"]}-#{firmware["status"]}",
+        :build        => firmware["build"],
+        :version      => firmware["version"],
+        :release_date => firmware["date"],
+      }
+    end
+
+    def parse_physical_server(node)
+      new_result = {
+        :type                   => ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalServer.name,
+        :name                   => node.name,
+        :ems_ref                => node.uuid,
+        :uid_ems                => @ems.uid_ems,
+        :hostname               => node.hostname,
+        :product_name           => node.productName,
+        :manufacturer           => node.manufacturer,
+        :machine_type           => node.machineType,
+        :model                  => node.model,
+        :serial_number          => node.serialNumber,
+        :field_replaceable_unit => node.FRU,
+        # Filled in later conditionally on what's available
+        :computer_system        => {:hardware => {:networks => [], :firmwares => []}}
+      }
+      new_result[:hardware] = get_hardwares(node)
```

Comment:
If you look at  https://github.com/ManageIQ/manageiq-providers-lenovo/pull/34    This assignment is now new_result[:computer_system][:hardware] = get_hardwares(node).  It may be worth rebasing this PR from 34 before it is merged.